### PR TITLE
DATAUP-765: assertify tests

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Convert existing `self.assert...` style-tests to the pytest-friendly `assert` versions.

Basically a big search-and-replace, aided by Ruff's auto-fix feature.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-765
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_ -- N/A

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook
- [ ] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: src/biokbase/narrative/jobs/appmanager.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 import random
 import re
 import traceback
-from typing import Callable, Optional
+from collections.abc import Callable
```
Comment (ialarmedalien) [inline src/biokbase/narrative/jobs/appmanager.py:9]: Recommended by Ruff

File: src/biokbase/narrative/tests/narrative_mock/mockclients.py [hunk 2]
```diff
@@ -326,9 +326,7 @@ def log_gen(log_params, total_lines=MAX_LOG_LINES):
             lines = []
             if skip < total_lines:
                 for i in range(total_lines - skip):
-                    lines.append(
-                        {"is_error": 0, "line": "This is line {}".format(i + skip)}
-                    )
+                    lines.append({"is_error": 0, "line": f"This is line {i + skip}"})
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/narrative_mock/mockclients.py:329]: use f-string

File: src/biokbase/narrative/tests/test_app_util.py [hunk 3]
```diff
@@ -74,7 +72,7 @@ def set_ws_name(ws_name):
 ]
 
 
-@pytest.mark.parametrize("result,path,expected", get_result_sub_path_cases)
+@pytest.mark.parametrize(("result", "path", "expected"), get_result_sub_path_cases)
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/test_app_util.py:75]: Ruff prefers tuples to comma-separated strings, obviously!

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseGenericSetViewer.js [hunk 4]
```diff
@@ -48,26 +48,32 @@ define([
             this._super(options);
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseGenericSetViewer.js:48]: These changes are back! 😱 
Was that intentional this time?

**Review by briehl — CHANGES_REQUESTED** [review]
Looks mostly fine, just a mix of the previous PR and some auto gen test updates.

Was the change to `kbaseGenericSetViewer.js` intended this time? If not, maybe excise it for the next go around.

Otherwise, please do the rebasing and it's ready.

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseGenericSetViewer.js (hunk 4 above)
Comment (ialarmedalien) [reply kbase-extension/static/kbase/js/widgets/function_output/kbaseGenericSetViewer.js:48]: Nope! Grrrrrr...

**Review by briehl — APPROVED** [review]
👍 
Thanks for removing that JS file change!

**Pull request merged**